tests/sql/operators/test_cleanup.py

```diff
@@ -204,3 +209,43 @@ def test_is_dag_running():
     assert not cleanup_op._is_dag_running(task_instances=task_instances)
     task_instances[0].state = State.RUNNING
     assert cleanup_op._is_dag_running(task_instances=task_instances)
+
+
+@pytest.mark.parametrize(
+    "executor,raise_error",
+    [
+        ("CeleryExecutor", False),
+        ("DebugExecutor", True),
+        ("SequentialExecutor", True),
+    ],
+)
+@mock.patch("astro.sql.operators.cleanup.CleanupOperator.get_all_task_outputs")
+def test_error_raised_with_blocking_op_executors(
+    mock_get_all_task_outputs, executor, raise_error
+):
+    """
+    Test that when SequentialExecutor or DebugExecutor is used an error is raised
+    if the other tasks are still running when cleanup is ran.
+
+    This is because Cleanup will block the entire thread and will cause deadlock until
+    execution_timeout of a task is reached
+    """
+    mock_get_all_task_outputs.return_value = []
+
+    dag = DAG(
+        "test_error_raised_with_blocking_op_executors", start_date=datetime(2022, 1, 1)
+    )
+    cleanup_task = CleanupOperator(dag=dag)
+    dr = DagRun(dag_id=dag.dag_id)
+    with mock.patch(
+        "astro.sql.operators.cleanup.CleanupOperator._is_dag_running"
+    ) as mock_is_dag_running, mock.patch.dict(
+        os.environ,
+        {"AIRFLOW__CORE__EXECUTOR": executor},
```

Comment:
Is this trying to set `env` while running?

Reply:
Yes